manager.go

```diff
@@ -8,57 +8,58 @@ import (
 	"github.com/libp2p/go-libp2p-core/peer"
 )
 
+// ValidationResult describes the result of validating a voucher
+type ValidationResult struct {
+	// Accepted indicates where the request was accepted if a request is not
+	// accepted, the request fails. This is true for revalidation as well
+	Accepted bool
+	// VoucherResult provides information to the other party about what happened
+	// with the voucher
+	VoucherResult
+	// LeaveRequestPaused indicates whether the request should stay paused
+	// even if the request was accepted
+	LeaveRequestPaused bool
+	// DataLimit specifies how much data this voucher is good for. When the amount
+	// amount data specified is reached (or shortly after), the request will pause
+	// pending revalidation. 0 indicates no limit.
+	DataLimit uint64
+	// RequiresFinalization indicates at the end of the transfer, the channel should
+	// be left open for a final settlement
+	RequiresFinalization bool
+}
+
 // RequestValidator is an interface implemented by the client of the
 // data transfer module to validate requests
 type RequestValidator interface {
 	// ValidatePush validates a push request received from the peer that will send data
+	// -- All information about the validation operation is contained in ValidationResult,
+	// including if it was rejected. Information about why a rejection occurred is embedded
+	// in the VoucherResult.
+	// -- error indicates something went wrong with the actual process of trying to validate
 	ValidatePush(
-		isRestart bool,
 		chid ChannelID,
 		sender peer.ID,
 		voucher Voucher,
 		baseCid cid.Cid,
-		selector ipld.Node) (VoucherResult, error)
+		selector ipld.Node) (ValidationResult, error)
 	// ValidatePull validates a pull request received from the peer that will receive data
+	// -- All information about the validation operation is contained in ValidationResult,
+	// including if it was rejected. Information about why a rejection occurred should be embedded
+	// in the VoucherResult.
+	// -- error indicates something went wrong with the actual process of trying to validate
 	ValidatePull(
-		isRestart bool,
```

Comment:
IIRC we need to know at the markets layer if the pull request is for a restart. Maybe check when you implement the markets layer changes and revisit.